src/containers/MasonryLayout.jsx

```diff
@@ -8,100 +7,71 @@ import PropTypes from 'prop-types';
  * @extends React.component
  */
 class MasonryLayout extends Component {
-  /**
- * @name defaultProps
- * @summary The default values of props
- */
-  static defaultProps = {
-    columnCount: 4,
-    gap: 20,
-  };
-
   /**
     * @name propTypes
     * @type {PropType}
     * @param {Object} propTypes - React PropTypes
     * @property {Object} items - React nodes or DOM elements to render
-    * @property {Number} columnCount - The number of columns of the masonry layout
-    * @property {Number} gap - The distance (in pixels) between columns and items
+    * @property {Boolean} showUserDetails - Whether or not to show activity owner details
+    * @property {Boolean} showLocation - Whether or not to show activity owner location
   */
   static propTypes = {
-    items: PropTypes.arrayOf(PropTypes.element).isRequired,
-    columnCount: PropTypes.number,
-    gap: PropTypes.number,
+    items: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
   };
 
-  componentWillMount() {
-    const columnWidth = this.getColumnWidth();
-    this.columns = Array(this.props.columnCount).fill({}).map((value, key) => (
-      <div
-        className='masonry__column'
-        style={{
-          width: columnWidth,
-          marginRight: `${this.props.gap}px`,
-        }}
-        ref={(el) => { this[`column${key}`] = el; }}
-        key={Math.random()}
-      />
-    ));
+  constructor(props) {
+    super(props);
+    this.state = {
+      items: [],
+    };
   }
 
-  componentDidMount() {
-    this.props.items.forEach((item, index) => {
-      // create a DOM element (wrapper) we can render to.
-      // this will wrap the item
-      const div = document.createElement('div');
-      div.style.marginBottom = `${this.props.gap}px`;
-      // render to the element
-      render(
-        item,
-        div,
-      );
-      // append the item to a column
-      this[`column${index % this.props.columnCount}`].append(div);
-    });
+  /**
+   * React component lifecycle method componentWillMount
+   * @memberof MasonryLayout
+   */
+  componentWillMount() {
+    this.setState({ items: this.props.items });
   }
 
   /**
-   * @name getColumnWidth
-   * @summary Computes width of a column
-   * @return {string} css value to set as width of a column
+   * React component lifecycle method componentWillReceiveProps
+   * @memberof MasonryLayout
+   * @param {Object} nextProps - props
    */
-  getColumnWidth = () => (
-    `calc(${100 / this.props.columnCount}% - ${this.props.gap}px)`
-  )
-
+  componentWillReceiveProps(nextProps) {
```

Comment:
Notes: Lets start replacing all these lifecycles with the new ones. Just coz its good practice and will save us some technical debt later on. Also we'll learn a lot in the process.
Here is a link to help out:
https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html